File: 049.rb
# 049
# Group Anagrams
# 2015-09-10
#####################################################
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
  dict = {}
  strs.each do |word|
    key = word.chars.sort.join
    dict[key] ||= []
    dict[key] << word
  end
  dict.each do |key, value|
    dict[key] = value.sort
  end
  dict.values
end

#####################################################

strs = ["eat", "tea", "tan", "ate", "nat", "bat"]

puts group_anagrams(strs).inspect

File: 049.py
#   049
#   Group Anagrams
#   2015-09-10
#####################################################
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dict = {}
        for word in strs:
            key = ''.join(sorted(word))
            if key not in dict:
                dict[key] = []
            dict[key].append(word)
        for key in dict:
            dict[key] = sorted(dict[key])
        return dict.values()

#####################################################

strs = ["eat", "tea", "tan", "ate", "nat", "bat"]

sol = Solution()

print sol.groupAnagrams(strs)

